pkg/daemon/update.go

```diff
@@ -975,6 +993,60 @@ func (dn *Daemon) updateOnClusterBuild(oldConfig, newConfig *mcfgv1.MachineConfi
 	return dn.reboot(fmt.Sprintf("Node will reboot into image %s / MachineConfig %s", newImage, newConfigName))
 }
 
+// Finalizes the revert process by enabling a special systemd unit prior to
+// rebooting the node.
+//
+// After we write the original factory image to the node, none of the files
+// specified in the MachineConfig will be written due to how rpm-ostree handles
+// file writes. Because those files are owned by the layered container image,
+// they are not present after reboot; even if we were to write them to the node
+// before rebooting. Consequently, after reverting back to the original image,
+// the node will lose contact with the control plane and the easiest way to
+// reestablish contact is to rebootstrap the node.
```

Comment:
Question: so by rebootstrap, you mean mostly to re-write the original contents, and keep the certificates around? i.e. the kubelet doesn't have to rebootstrap using its bootstrap creds which may lead to a new CSR?

Reply:
I mean the former case where we just need to rewrite the original contents.